atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/async/client/creation/DefaultCqlClientFactory.java

```diff
@@ -62,22 +63,22 @@ public DefaultCqlClientFactory() {
         this(Cluster::builder);
     }
 
-    public CqlClient constructClient(
+    public Optional<CqlClient> constructClient(
             TaggedMetricRegistry taggedMetricRegistry,
             CassandraKeyValueServiceConfig config,
             boolean initializeAsync) {
-        return config.servers().accept(new CassandraServersConfigs.Visitor<CqlClient>() {
+        return config.servers().accept(new CassandraServersConfigs.Visitor<Optional<CqlClient>>() {
             @Override
-            public CqlClient visit(CassandraServersConfigs.DefaultConfig defaultConfig) {
-                return ThrowingCqlClientImpl.SINGLETON;
+            public Optional<CqlClient> visit(CassandraServersConfigs.DefaultConfig defaultConfig) {
+                return Optional.empty();
             }
 
             @Override
-            public CqlClient visit(CassandraServersConfigs.CqlCapableConfig cqlCapableConfig) {
+            public Optional<CqlClient> visit(CassandraServersConfigs.CqlCapableConfig cqlCapableConfig) {
                 if (!cqlCapableConfig.validateHosts()) {
                     log.warn("Your CQL capable config is wrong, the hosts for CQL and Thrift are not the same, using "
                             + "async API will result in an exception.");
```

Comment:
nit: This log message is no longer accurate after the change (we can now say using the async API may result in undefined behaviour: I know that officially we will use the sync API, but I don't think we necessarily need to guarantee that).

Reply:
Changed.